### PR TITLE
Adding depth in case of git config in fetch section of app/package

### LINKED ISSUE #1632: Consume depth flag of vendir in git fetch section of app/package spec
**Describe the problem/challenge you have**
In the gitops scenario, if the git repo is very huge it takes lot of time in the fetch stage. To make things a bit faster, instead of fetching all the commit history of git, by default, we can fetch only current commit.  

**Describe the solution you'd like**
Add `depth` as the field in git config of the fetch section and set its value to `1` by default in the code. 

**Anything else you would like to add:**
[Additional information that will assist in solving the issue.]

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by rohitagg2020**
Adding depth in case of git config in fetch section of app/package

Fixes #1632 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/apis/kappctrl/v1alpha1/types_fetch.go [hunk 1]
```diff
@@ -109,6 +109,9 @@ type AppFetchGit struct {
 	// Force the usage of HTTP Basic Auth when Basic Auth is provided (optional)
 	// +optional
 	ForceHTTPBasicAuth bool `json:"forceHTTPBasicAuth,omitempty" protobuf:"varint,7,opt,name=forceHTTPBasicAuth"`
+	// depth of commits to fetch; 1 (default) means current, 0 means everything (optional)
```
Comment (praveenrewar) [inline pkg/apis/kappctrl/v1alpha1/types_fetch.go]: nit
```suggestion
	// depth of commits to fetch; 1 (default) means only latest commit, 0 means everything (optional)
```

File: pkg/fetch/vendir.go [hunk 2]
```diff
@@ -160,6 +161,13 @@ func (v *Vendir) httpConf(http v1alpha1.AppFetchHTTP) vendirconf.DirectoryConten
 }
 
 func (v *Vendir) gitConf(git v1alpha1.AppFetchGit) vendirconf.DirectoryContents {
+	var depth int
+	if git.Depth == nil {
+		depth = 1
+	} else {
+		depth = int(*(git.Depth))
+	}
+
```
Comment (praveenrewar) [inline pkg/fetch/vendir.go]:
```suggestion
	depth := 1
	if git.Depth != nil {
		depth = int(*(git.Depth))
	}
```

File: test/e2e/kappcontroller/packageinstall_test.go [hunk 3]
```diff
@@ -624,13 +624,19 @@ spec:
 		}
 
 		tmpl := cr.Spec.Template[0]
-		if tmpl.Ytt != nil && len(tmpl.Ytt.Paths) != 0 {
-			if tmpl.Ytt.Paths[0] != "file.yml" {
-				t.Fatalf("\nExpected App spec.template.ytt.paths to contain file.yml\nGot: %s", tmpl.Ytt.Paths[0])
+
+		if tmpl.Ytt != nil {
```
Comment (praveenrewar) [inline test/e2e/kappcontroller/packageinstall_test.go]: I am not sure how this change is related to the rest of the PR 🤔

Comment (rohitagg2020) [reply test/e2e/kappcontroller/packageinstall_test.go]: There were test case failing while running locally. Hence, added this change.

File: pkg/fetch/vendir.go (hunk 2 above)
Comment (rohitagg2020) [reply pkg/fetch/vendir.go]: Adopted

**Review by praveenrewar — APPROVED** [review]
LGTM!

**Pull request merged**